File: frontend/src/mappers/contactMapper.js
export function mapContact(data) {
    if (!data) {
        throw new Error("Contact data is missing");
    }

    return {
        name: data.name ?? "",
        phone: data.phone ?? "",
        email: data.email ?? "",
        location: data.location ?? "",
        githubUrl: data.githubUrl ?? "",
        linkedInUrl: data.linkedInUrl ?? "",
        instagramUrl: data.instagramUrl ?? "",
        spotifyUrl: data.spotifyUrl ?? "",
        creditText: data.creditText ?? "",
    };
}

File: backend/Application/Mappers/ContactMapper.cs
using Application.DTOs.Responses;
using Application.Interfaces;
using Domain.Entities;

namespace Application.Mappers;

public class ContactMapper : IContactMapper
{
    public ContactResponse Map(Contact contact)
    {
        return new ContactResponse
        {
            Name = contact.Name,
            Phone = contact.Phone,
            Email = contact.Email,
            Location = contact.Location,
            GitHubUrl = contact.GitHubUrl,
            LinkedInUrl = contact.LinkedInUrl,
            InstagramUrl = contact.InstagramUrl,
            SpotifyUrl = contact.SpotifyUrl,
            CreditText = contact.CreditText
        };
    }
}